src/lib/sys/configuration.cpp

```diff
@@ -741,6 +755,13 @@ bool Configuration::saveConfig(bool restart) {
                 if(cbor_value_get_string_length(&it,&size) != CborNoError){
                     return false;
                 }
+                break;
+            }
+            case ARRAY: {
```

Comment:
We should also update `topology_sampler.cpp` : `encode_cbor_configuration` to support array types!

Reply:
Thanks for the tip! It didn't break the build with a "switch must handle all enum values" so I didn't see it before. Done in commit d7d621d. It was a little tricky since, in addition to knowing how to calculate on the internal pointer of the encoder like last night, I also had to find the internal pointer of the parser.